File: packages/node/src/resources/health.ts
import type { MissionInbox } from '../client.js';
import type { HealthStatus } from '../types.js';

/**
 * The `health` resource. Access via `mi.health`.
 *
 * The health endpoint is unauthenticated and returns a small plain-text or
 * JSON response indicating the API is reachable. Useful as a liveness probe.
 */
export class Health {
  /** @internal */
  constructor(private readonly client: MissionInbox) {}

  /** Ping the API. Resolves with the API's health string on success. */
  async check(): Promise<HealthStatus> {
    return this.client.request<HealthStatus>({
      method: 'GET',
      path: '/api/health',
    });
  }
}

File: packages/python/src/missioninbox/resources/health.py
"""The `health` resource. Access via `mi.health`."""

from __future__ import annotations

from typing import Union

from .._http import HttpClient


class Health:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def check(self) -> Union[str, dict]:
        """Ping the API. Returns the API's health string (or dict, if JSON)."""
        return self._http.request("GET", "/api/health")
